### PR TITLE
Use REQUIRED constant instead of None for options

### PR DESCRIPTION
**Pull request by siminm**
Sometimes None is a valid default value, and it should not be reserved
to identify a required parameter. This change converts all actors to use
a REQUIRED identifier (meta class).

### CODE REVIEW

File: kingpin/actors/base.py [hunk 1]
```diff
@@ -132,10 +132,17 @@ def _validate_options(self):
         """
 
         # Loop through all_options, and find the required ones
-        # Required options have `None` as their default value.
+        self.log.debug(self.all_options.items())
+        if len(self.all_options.items()):
```
Comment (diranged) [inline kingpin/actors/base.py]: Whats going on here? The code is simple -- but I don't actually know what its doing.

Comment (siminm) [inline kingpin/actors/base.py]: Oh goodness. That wasn't supposed to make the commit.

**Comment by siminm** [comment]
Removed the extra debug lines.

**Comment by diranged** [comment]
:+1:

**Pull request merged**